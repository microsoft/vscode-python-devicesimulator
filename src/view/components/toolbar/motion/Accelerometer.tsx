import * as React from "react";
import { SENSOR_LIST } from "../../../constants";
import { ISensorProps, ISliderProps } from "../../../viewUtils";
import { ThreeDimensionSlider } from "./threeDimensionSlider/ThreeDimensionSlider";

const MOTION_SLIDER_PROPS_X: ISliderProps = {
    axisLabel: "X",
    maxLabel: "Right",
    maxValue: 1023,
    minLabel: "Left",
    minValue: -1023,
    type: SENSOR_LIST.MOTION_X,
};
const MOTION_SLIDER_PROPS_Y: ISliderProps = {
    axisLabel: "Y",
    maxLabel: "Front",
    maxValue: 1023,
    minLabel: "Back",
    minValue: -1023,
    type: SENSOR_LIST.MOTION_Y,
};
const MOTION_SLIDER_PROPS_Z: ISliderProps = {
    axisLabel: "Z",
    maxLabel: "Down",
    maxValue: 1023,
    minLabel: "Up",
    minValue: -1023,
    type: SENSOR_LIST.MOTION_Z,
};

const MOTION_SENSOR_PROPERTIES: ISensorProps = {
    LABEL: "Motion sensor",
    sliderProps: [
        MOTION_SLIDER_PROPS_X,
        MOTION_SLIDER_PROPS_Y,
        MOTION_SLIDER_PROPS_Z,
    ],
    unitLabel: "Lux",
};
interface IProps {
    axisValues: {
        X_AXIS: number;
        Y_AXIS: number;
        Z_AXIS: number;
    };
    onUpdateValue: (sensor: SENSOR_LIST, value: number) => void;
}
export const Accelerometer: React.FC<IProps> = (props: IProps) => {
    return (
        <div className="AccelerometerBar">
            <br />
            {/* Implement Gestures Here */}
            <ThreeDimensionSlider
                axisProperties={MOTION_SENSOR_PROPERTIES}
                onUpdateValue={props.onUpdateValue}
                axisValues={props.axisValues}
            />
        </div>
    );
};
