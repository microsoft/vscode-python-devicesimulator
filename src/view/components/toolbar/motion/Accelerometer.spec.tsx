import * as React from "react";
import * as ReactDOM from "react-dom";
import { IntlProvider } from "react-intl";
import * as testRenderer from "react-test-renderer";
import { Accelerometer } from "./Accelerometer";

describe("Accelerometer component ", () => {
    const mockProps = {
        axisValues: {
            X_AXIS: 1,
            Y_AXIS: 0,
            Z_AXIS: 1,
        },
        onUpdateValue: () => {},
    };

    it("should render correctly", () => {
        const component = testRenderer
            .create(
                <IntlProvider locale="en">
                    <Accelerometer
                        axisValues={mockProps.axisValues}
                        onUpdateValue={mockProps.onUpdateValue}
                    />
                </IntlProvider>
            )
            .toJSON();
        expect(component).toMatchSnapshot();
    });

    it("should render without crashing", () => {
        const div = document.createElement("div");
        ReactDOM.render(
            <IntlProvider locale="en">
                <Accelerometer
                    axisValues={mockProps.axisValues}
                    onUpdateValue={mockProps.onUpdateValue}
                />
            </IntlProvider>,
            div
        );
        ReactDOM.unmountComponentAtNode(div);
    });
});
