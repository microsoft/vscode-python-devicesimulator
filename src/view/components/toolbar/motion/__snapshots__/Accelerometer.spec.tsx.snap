// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accelerometer component  should render correctly 1`] = `
<div
  className="accelerometer"
>
  <div
    className="GenericSliderComponent"
  >
    <div
      className="inputSlider"
    >
      <span>
        X
      </span>
      <input
        aria-label="motion_x sensor input X"
        className="sliderValue"
        defaultValue="-1,023"
        onInput={[Function]}
        onKeyUp={[Function]}
        pattern="^-?[0-9]{0,4}$"
        type="text"
        value={1}
      />
      <span
        className="sliderArea"
      >
        <span
          className="upLabelArea"
        >
          <span
            className="minLabel"
          >
            -1023
          </span>
          <span
            className="maxLabel"
          >
            1023
          </span>
        </span>
        <input
          aria-label="motion_x sensor"
          aria-valuemax={1023}
          aria-valuemin={-1023}
          aria-valuenow={1}
          className="slider"
          defaultValue="-1,023"
          disabled={false}
          max={1023}
          min={-1023}
          onChange={[Function]}
          onKeyUp={[Function]}
          onMouseUp={[Function]}
          type="range"
          value={1}
        />
        <span
          className="downLabelArea"
        >
          <span
            className="minLabel"
          >
            Left
          </span>
          <span
            className="maxLabel"
          >
            Right
          </span>
        </span>
      </span>
    </div>
    <br />
    <div
      className="inputSlider"
    >
      <span>
        Y
      </span>
      <input
        aria-label="motion_y sensor input Y"
        className="sliderValue"
        defaultValue="-1,023"
        onInput={[Function]}
        onKeyUp={[Function]}
        pattern="^-?[0-9]{0,4}$"
        type="text"
        value={0}
      />
      <span
        className="sliderArea"
      >
        <span
          className="upLabelArea"
        >
          <span
            className="minLabel"
          >
            -1023
          </span>
          <span
            className="maxLabel"
          >
            1023
          </span>
        </span>
        <input
          aria-label="motion_y sensor"
          aria-valuemax={1023}
          aria-valuemin={-1023}
          aria-valuenow={0}
          className="slider"
          defaultValue="-1,023"
          disabled={false}
          max={1023}
          min={-1023}
          onChange={[Function]}
          onKeyUp={[Function]}
          onMouseUp={[Function]}
          type="range"
          value={0}
        />
        <span
          className="downLabelArea"
        >
          <span
            className="minLabel"
          >
            Back
          </span>
          <span
            className="maxLabel"
          >
            Front
          </span>
        </span>
      </span>
    </div>
    <br />
    <div
      className="inputSlider"
    >
      <span>
        Z
      </span>
      <input
        aria-label="motion_z sensor input Z"
        className="sliderValue"
        defaultValue="-1,023"
        onInput={[Function]}
        onKeyUp={[Function]}
        pattern="^-?[0-9]{0,4}$"
        type="text"
        value={1}
      />
      <span
        className="sliderArea"
      >
        <span
          className="upLabelArea"
        >
          <span
            className="minLabel"
          >
            -1023
          </span>
          <span
            className="maxLabel"
          >
            1023
          </span>
        </span>
        <input
          aria-label="motion_z sensor"
          aria-valuemax={1023}
          aria-valuemin={-1023}
          aria-valuenow={1}
          className="slider"
          defaultValue="-1,023"
          disabled={false}
          max={1023}
          min={-1023}
          onChange={[Function]}
          onKeyUp={[Function]}
          onMouseUp={[Function]}
          type="range"
          value={1}
        />
        <span
          className="downLabelArea"
        >
          <span
            className="minLabel"
          >
            Up
          </span>
          <span
            className="maxLabel"
          >
            Down
          </span>
        </span>
      </span>
    </div>
    <br />
  </div>
</div>
`;
